index.js

```diff
@@ -1944,11 +1979,16 @@
     });
   }
 
-  //  invalidArgumentsLength :: (String, Integer, Integer) -> Error
-  function invalidArgumentsLength(name, expectedLength, actualLength) {
-    return new TypeError(q(name) +
-                         ' requires ' + numArgs(expectedLength) + ';' +
-                         ' received ' + numArgs(actualLength));
+  //  tooManyArguments :: (TypeInfo, Integer) -> Error
```

Comment:
This is a bit confusing. Currently we have two functions:

  - `invalidArgumentsLength`, used when a `def`'d function itself is misapplied; and
  - `invalidArgumentsLength_`, used when a function *argument* is misapplied.

The difference is best explained by example:

```javascript
const $ = require('.');

const a = $.TypeVariable('a');
const b = $.TypeVariable('b');

const def = $.create({checkTypes: true, env: $.env});

//    map :: (a -> b) -> Array a -> Array b
const map =
def('map',
    {},
    [$.Function([a, b]), $.Array(a), $.Array(b)],
    (f, xs) => xs.map(f));

map(Math.sqrt, [], [], [], []);
// ! TypeError: ‘map’ requires two arguments; received five arguments

map(Math.sqrt, [1, 4, 9]);
// ! TypeError: ‘map’ applied ‘a -> b’ to the wrong number of arguments
//
//   map :: (a -> b) -> Array a -> Array b
//           ^
//           1
//
//   Expected one argument but received three arguments:
//
//     - 1
//     - 0
//     - [1, 4, 9]
```

`invalidArgumentsLength` is currently referenced in two places.

In `curry`:

```javascript
var delta = _indexes.length - arguments.length;
if (delta < 0) {
  throw invalidArgumentsLength(typeInfo.name, n, n - delta);
}
```

In `def`:

```javascript
if (arguments.length !== def.length) {
  throw invalidArgumentsLength('def', def.length, arguments.length);
}
```

Now that `$.def` is defined via `def`, the latter is no longer necessary. In `curry` we only ever invoke `invalidArgumentsLength` when we receive too *many* arguments, so we can make the error message more specific (and change the function's name from `invalidArgumentsLength` to `tooManyArguments`).

If we're to rename `invalidArgumentsLength`, the `_` in `invalidArgumentsLength_` is no longer necessary. This pull request makes the following changes:

  - `invalidArgumentsLength` → `tooManyArguments`; and
  - `invalidArgumentsLength_` → `invalidArgumentsLength`.

Have I done a reasonable job of making this confusing situation clearer?